baseband/vdif/header.py

```diff
@@ -28,6 +32,41 @@
                   precision=9)
 
 
+_vdif_edv_header_classes = {}
+"""Dict for storing pairs of VDIF EDV and header classes."""
+
+class _VDIFHeaderRegistry(type):
+    """
+    Registry of VDIF Header EDV types, using the ``_vdif_edv_header_classes``
+    dict.  Checks for keyword and subclass conflicts before registering.
+    """
+    _vdif_edv_header_classes = _vdif_edv_header_classes
+
+    def __init__(cls, name, bases, dct):
+
+        # Check that class is subclass of VDIFHeader
+        if not issubclass(cls, VDIFHeader):
+            raise TypeError(cls.__name__ + " is not a subclass of VDIFHeader!")
+
+        # Extract edv from class; convert to -1 if edv == False (VDIFLegacy)
+        edv = cls._edv
+        if edv is False:
+            edv = -1
+
+        # Check if EDV is already registered (edv == None is invalid).
+        if edv is None:
+            raise ValueError("edv cannot be None!")
+        elif edv in _VDIFHeaderRegistry._vdif_edv_header_classes:
+            raise ValueError("EDV == " + str(cls._edv) + " already registered"
+                " in _VDIFHeaderRegistry._vdif_edv_header_classes.keys() = " +
+                str(_VDIFHeaderRegistry._vdif_edv_header_classes.keys()) + 
+                "!  (EDV == False is converted to -1 in registry.)")
+
+        _VDIFHeaderRegistry._vdif_edv_header_classes.update({edv: cls})
+        
+        super(_VDIFHeaderRegistry, cls).__init__(name, bases, dct)
```

Comment:
I'm pretty sure cls._edv is set in type.__new__, while type.__init__ doesn't do anything for class creation.